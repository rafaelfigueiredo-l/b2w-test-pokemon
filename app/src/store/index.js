import { createStore } from "redux";

const INITIAL_STATE = {
  theme: {},
  pokemon: [],
};
function course(state = INITIAL_STATE, action) {
  switch (action.type) {
    case "SET_THEME":
      return {
        ...state,
        theme: action.payload,
      };

    case "SET_POKEMON":
      return {
        ...state,
        pokemon: action.payload,
      };

    default:
      return state;
  }
}

const store = createStore(
  course,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

export default store;
