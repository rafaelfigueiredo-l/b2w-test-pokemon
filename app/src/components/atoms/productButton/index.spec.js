import { Provider } from "react-redux";
import store from "../../../store";
import ProductButton from "./";

describe("<ProductButton />", () => {
  const component = (
    <Provider store={store}>
      <ProductButton />
    </Provider>
  );

  const mounted = mount(component);

  it("should mount correctly ProductButton component", () => {
    expect(mounted.find(".productButton")).toHaveLength(2);
    expect(mounted).toMatchSnapshot();
  });

  it("should unmount correctly ProductButton component", () => {
    expect(mounted.unmount()).toEqual({});
  });
});
