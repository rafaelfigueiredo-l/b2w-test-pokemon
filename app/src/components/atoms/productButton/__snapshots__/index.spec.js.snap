// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductButton /> should mount correctly ProductButton component 1`] = `
<styled.div
  className="productButton"
>
  <div
    className="sc-AxjAm bkRpRN productButton"
  >
    <button
      onClick={[Function]}
    >
      COMPRAR
    </button>
  </div>
</styled.div>
`;
